viewer/src/tools/AxisView/AxisViewTool.ts

```diff
@@ -0,0 +1,321 @@
+    context.fill();
+
+    if (faceConfig.label !== null) {
+      const fontSize = faceConfig.fontSize ?? textureSize / 3;
+      context.font = `bold ${fontSize}px Arial`;
+      context.textAlign = 'center';
+      context.fillStyle = faceConfig.fontColor!.getStyle();
+      context.fillText(faceConfig.label!, textureSize / 2, textureSize / 2 + fontSize / 3);
+    }
+
+    return new THREE.CanvasTexture(canvas);
+  }
+
+  private createBoxFace(position: THREE.Vector3, faceConfig: AxisBoxFaceConfig, upVector = new THREE.Vector3(0, 1, 0)) {
+    const face = new THREE.Mesh(
+      this._boxFaceGeometry,
+      new THREE.MeshBasicMaterial({ map: this.getFaceTexture(faceConfig, this._layoutConfig.size!) })
+    );
+
+    face.position.copy(position.multiplyScalar(0.5 * this._boxFaceGeometry.parameters.width));
+    face.lookAt(position.multiplyScalar(2));
+
+    face.userData.upVector = upVector;
+
+    return face;
+  }
+
+  private moveCameraTo(
+    camera: THREE.PerspectiveCamera,
+    cameraControls: ComboControls,
+    targetAxis: THREE.Vector3,
+    targetUpAxis: THREE.Vector3,
+    duration: number
+  ) {
+    const currentCameraPosition = camera.position.clone();
+    const cameraTarget = cameraControls.getState().target;
+
+    const targetRelativeStartPosition = currentCameraPosition.clone().sub(cameraTarget);
+    const radius = targetRelativeStartPosition.length();
+
+    const normalizedFrom = targetRelativeStartPosition.clone().normalize();
+
+    const omega = Math.acos(normalizedFrom.dot(targetAxis));
+
+    const from = { t: 0 };
+    const to = { t: 1 };
+
+    const animation = new TWEEN.Tween(from);
+
+    const forward = targetAxis.clone();
+
+    const fromRotation = camera.quaternion.clone();
+    const toRotation = new THREE.Quaternion().setFromRotationMatrix(
+      new THREE.Matrix4().makeBasis(targetUpAxis.clone().cross(forward), targetUpAxis, forward)
+    );
+
+    const epsilon = 1e-6;
+
+    if (fromRotation.angleTo(toRotation) < epsilon) {
+      return;
+    }
+
+    const tmpPosition = new THREE.Vector3();
+    const tmpRotation = new THREE.Quaternion();
+    const tween = animation
+      .to(to, duration)
+      .onUpdate(() => {
+        tmpPosition
+          .copy(normalizedFrom)
+          .multiplyScalar(Math.sin((1 - from.t) * omega) / Math.sin(omega))
+          .add(targetAxis.clone().multiplyScalar(Math.sin(from.t * omega) / Math.sin(omega)));
+
+        tmpPosition.multiplyScalar(radius);
+        tmpPosition.add(cameraTarget);
+
+        THREE.Quaternion.slerp(fromRotation, toRotation, tmpRotation, from.t);
+
+        camera.position.copy(tmpPosition);
+        camera.setRotationFromQuaternion(tmpRotation);
+      })
+      .start(TWEEN.now())
+      .onStart(() => {
+        cameraControls.enabled = false;
+      })
+      .onComplete(() => {
+        cameraControls.setState(camera.position, cameraTarget);
+        cameraControls.enabled = true;
+      });
```

Comment:
Store `enabled` before tweening and reset on completion, rather than explicitly setting to true.